homeassistant/components/media_player/pandora.py

```diff
@@ -0,0 +1,232 @@
+        self.update_ha_state()
+
+    def turn_off(self):
+        """Turn the media player off."""
+        self._pianobar_remote.stop()
+        self._player_state = STATE_OFF
+        self.update_ha_state()
+
+    def media_play(self):
+        """Send play command."""
+        self._pianobar_remote.send_service(SERVICE_MEDIA_PLAY_PAUSE)
+        self._player_state = STATE_PLAYING
+        self.update_ha_state()
+
+    def media_pause(self):
+        """Send pause command."""
+        self._pianobar_remote.send_service(SERVICE_MEDIA_PLAY_PAUSE)
+        self._player_state = STATE_PAUSED
+        self.update_ha_state()
+
+    def media_next_track(self):
+        """Go to next track."""
+        self._pianobar_remote.send_service(SERVICE_MEDIA_NEXT_TRACK)
+        self.update_ha_state()
+
+    @property
+    def supported_media_commands(self):
+        """Show what this supports."""
+        return PANDORA_SUPPORT
+
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self._pianobar_remote.station
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        self._pianobar_remote.get_stations()
+        return self._pianobar_remote.stations
+
+    def select_source(self, source):
+        """Select input source."""
+        self._pianobar_remote.set_station(source)
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        self._pianobar_remote.update_playing()
+        return self._pianobar_remote.song
+
+    @property
+    def media_artist(self):
+        """Artist of current playing media, music track only."""
+        return self._pianobar_remote.artist
+
+    @property
+    def media_album_name(self):
+        """Album name of current playing media, music track only."""
+        return self._pianobar_remote.album
+
+
+class PianobarRemote(object):
+    """
+    A light interface to the Pianobar remote control.
+
+    pianobar has a FIFO remote control system that is challenging to
+    get working if pianobar is in a subprocess. Using pexpect is more stable
+    so this implementation uses pexpect.
+
+    """
+
+    CMD_MAP = {SERVICE_MEDIA_NEXT_TRACK: 'n',
+               SERVICE_MEDIA_PLAY_PAUSE: 'p',
+               SERVICE_MEDIA_PLAY: 'p',
+               SERVICE_VOLUME_UP: ')',
+               SERVICE_VOLUME_DOWN: '('}
+
+    CURRENT_SONG_PATTERN = re.compile(r'"(.*?)"\s+by\s+"(.*?)"\son\s+"(.*?)"')
+    STATION_PATTERN = re.compile('Station "(.+?)"')
+    TIME_PATTERN = re.compile(rb'\d\d:\d\d')
+
+    def __init__(self):
+        """Construct a pianobar remote controller."""
+        self._pianobar = None
+        self.station = ''
+        self.song = ''
+        self.artist = ''
+        self.album = ''
+        self.stations = []
+
+    def start(self):
+        """Start the pianobar subprocess."""
+        import pexpect
+        self._pianobar = pexpect.spawn('sudo -u pi pianobar')
```

Comment:
Derp. Yeah that's a big mistake. I'll fix and just let the user specify where the config file is.